src/coreclr/jit/emitxarch.cpp

```diff
@@ -487,12 +487,6 @@ bool emitter::IsFlagsAlwaysModified(instrDesc* id)
 //    true if previous instruction zeroed reg's upper 32 bits.
 //    false if it did not, or if we can't safely determine.
 //
-// Notes:
-//    Currently only looks back one instruction.
-//
-//    movsx eax, ... might seem viable but we always encode this
-//    instruction with a 64 bit destination. See TakesRexWPrefix.
-
```

Comment:
I think both `emitter::AreUpper32BitsZero()` and `emitter::AreUpper32BitsSignExtended` need to be under `#ifdef TARGET_64BIT`. They simply don't make sense otherwise.

Reply:
I have another PR that does this: https://github.com/dotnet/runtime/pull/82276